jupyter-extension/jupyterlab_pachyderm/pps_client.py

```diff
@@ -70,32 +97,42 @@ async def create(self, path: str, config: dict):
 @dataclass
 class PpsConfig:
 
+    notebook_path: Path
     pipeline_name: str
     image: str
     requirements: Optional[str]
     input_spec: dict  # We may be able to use the pachyderm SDK to parse and validate.
 
     @classmethod
-    def from_json(cls, config) -> "PpsConfig":
-        """Parses a config from a json object.
+    def from_notebook(cls, notebook_path: Union[str, Path]) -> "PpsConfig":
+        """Parses a config from the metadata of a notebook file.
 
-        Raises HTTPError with 500 code if required field is not specified.
+        Raises ValueError if required field is not specified.
         """
-        pipeline_name = config.get('pipeline_name', None)
+        notebook_path = Path(notebook_path)
+        notebook_data = json.loads(notebook_path.read_bytes())
+
+        config = notebook_data['metadata'].get('same_config')
```

Comment:
We should figure out a way to share the config key name so the FE and BE use the same value

Reply:
future work note